packages/framework/src/errors/resource.errors.ts

```diff
@@ -14,14 +14,9 @@ export abstract class ResourceNotFoundError extends NotFoundError {
   }
 }
 
-export abstract class ResourceExecutionFailed extends InternalServerError {
-  constructor(resource: ResourceEnum, id: string) {
-    super(`Failed to execute ${toPascalCase(resource)} with id: \`${id}\`. Please try again later.`);
-  }
-}
-
-export abstract class ResourcePreviewFailed extends InternalServerError {
```

Comment:
We don't need a separate class for preview errors, we just use the `Execution` errors abstract class with different `action` injected.